cpp/benchmarks/fixture/benchmark_fixture.hpp

```diff
@@ -31,7 +31,9 @@ inline auto make_cuda() { return std::make_shared<rmm::mr::cuda_memory_resource>
 
 inline auto make_pool()
 {
-  return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda());
+  static std::shared_ptr<rmm::mr::device_memory_resource> mr;
+  if (!mr) { mr = rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda()); }
+  return mr;
```

Comment:
Would this work too?

```suggestion
  static std::shared_ptr<rmm::mr::device_memory_resource> mr = 
     rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(make_cuda());
  return mr;
```

It seems the `inline` would make this static per fixture and not per binary. I think you can have multiple fixtures per binary.  Does `make_pool` have to be inline?

Reply:
This is what you want.
```suggestion
  static rmm::mr::cuda_memory_resource cuda_mr;
  static rmm::mr::pool_memory_resource pool_mr{&cuda_mr};
  return pool_mr;
```

Even with `inline`, the same pool will be shared among all TUs. I tested this extensively as it is how the RMM `per_device_resource` works. See: https://github.com/jrhemstad/link_test